tests/rptest/clients/golang_serde_client.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2023 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+import logging
+import json
+import shlex
+import subprocess
+
+from uuid import uuid4
+
+from rptest.clients.serde_client_utils import SchemaType
+
+
+class GolangSerdeClient:
+    def __init__(self,
+                 brokers,
+                 schema_registry_url,
+                 schema_type: SchemaType,
+                 *,
+                 topic=str(uuid4()),
+                 group=str(uuid4()),
+                 logger=logging.getLogger("GolangSerdeClient"),
+                 security_config: dict = None):
+
+        self.logger = logger
+        self.brokers = brokers
+        self.schema_registry_url = schema_registry_url
+        self.schema_type = schema_type
+        self.topic = topic
+        self.group = group
+        self.security_config = security_config
+
+    def _exec(self, cmd: str):
+        self.logger.info(f"Starting golang serde client with command: {cmd}")
+
+        try:
+            process = subprocess.Popen(shlex.split(cmd),
```

Comment:
Same thing here as in the java client. Consider running this on a ducktape node.